pkg/agent/clean/binding.go

```diff
@@ -47,19 +48,23 @@ type bindingsCleanup struct {
 	roleBindings        v1.RoleBindingClient
 }
 
-func Bindings() error {
+func Bindings(clientConfig *restclient.Config) error {
 	logrus.Info("Starting bindings cleanup")
 	if os.Getenv("DRY_RUN") == "true" {
 		logrus.Info("DRY_RUN is true, no objects will be deleted/modified")
 		dryRun = true
 	}
-
-	config, err := clientcmd.BuildConfigFromFlags("", os.Getenv("KUBECONFIG"))
-	if err != nil {
-		logrus.Errorf("Error in building the cluster config %v", err)
-		return err
+	var config *restclient.Config
+	var err error
+	if clientConfig == nil {
+		config, err = clientcmd.BuildConfigFromFlags("", os.Getenv("KUBECONFIG"))
+		if err != nil {
+			logrus.Errorf("Error in building the cluster config %v", err)
+			return err
+		}
+	} else {
+		config = clientConfig
```

Comment:
Swap the if else around so if we have a config use it, otherwise fall through to trying to build it